DANS-KNAW/easy-ingest-dispatcher
# New dispatch monitor

~~fixes EASY-~~

#### When applied it will
* introduce a simplified but better working file system monitor
* also ingest deposits that were put in the `/data/easy-sword2/deposits` folder before the service was started

#### Where should the reviewer @DANS-KNAW/easy start?
Please make **absolutely sure** you understand what is going on in this code. I know we haven't done Rx in the workshops yet, but this should be fairly simple to read.

#### How should this be manually tested?
* `git checkout` the pull request
* `mvnci`
* deploy role on deasy
* stop the service using `sudo service easy-ingest-dispatcher stop` (you should see corresponding events in the log)
* put one or more deposits in the `/data/easy-sword2/deposits` folder
* start the service again using `sudo service easy-ingest-dispatcher start`
* the service should now start the ingest procedure on these deposits
* add more deposits to the same folder (during the previous ingest procedure and/or after it's done) and watch them being ingested as well
* try to stop the service while it is processing elements and you'll see that it will first process all remaining deposits before shutting down
